src/components/heading/heading.test.tsx

```diff
@@ -0,0 +1,239 @@
+  ["h5", 5],
+] as [HeadingType, number][])(
+  "renders the correct heading element when the `headingType` prop is passed as %s",
+  (headingType, level) => {
+    render(<Heading title="foo" headingType={headingType} />);
+
+    const heading = screen.getByRole("heading", { level });
+    expect(heading).toBeVisible();
+  }
+);
+
+test("renders wih a custom help text, via the `help` prop", async () => {
+  render(
+    <Heading title="foo" help="help text" helpAriaLabel="help-aria-label" />
+  );
+
+  const user = userEvent.setup();
+  const helpIconButton = screen.getByRole("button", {
+    name: "help-aria-label",
+  });
+  await user.hover(helpIconButton);
+  const helpText = await screen.findByText("help text");
+
+  expect(helpIconButton).toHaveAttribute("data-element", "help");
+  expect(helpText).toBeVisible();
+});
+
+test("renders wih a custom help text link, via the `helpLink` prop", () => {
+  render(
+    <Heading
+      title="foo"
+      help="help text"
+      helpAriaLabel="help-aria-label"
+      helpLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const helpIconLink = screen.getByRole("link", { name: "help-aria-label" });
+  expect(helpIconLink).toHaveAttribute(
+    "href",
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("renders wih an 'aria-label' attribute on the help, via the `aria-label` prop", () => {
+  render(
+    <Heading
+      title="foo"
+      help="help text"
+      helpAriaLabel="help-aria-label"
+      helpLink="https://www.warnerbros.com/movies/heat"
+    />
+  );
+
+  const helpIconLink = screen.getByRole("link", { name: "help-aria-label" });
+  expect(helpIconLink).toBeVisible();
+});
+
+test("renders a back link, when `backLink` prop is a href", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  expect(backLink).toHaveAttribute(
+    "href",
+    "https://www.warnerbros.com/movies/heat"
+  );
+});
+
+test("focuses the back link, when `backLink` is a href on mouse down", () => {
+  render(
+    <Heading title="foo" backLink="https://www.warnerbros.com/movies/heat" />
+  );
+
+  const backLink = screen.getByRole("link", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+test("renders a back button, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  expect(backLink).toBeVisible();
+});
+
+test("focuses the back link, when `backLink` prop is a function", () => {
+  const backLinkFunction = jest.fn();
+  render(<Heading title="foo" backLink={backLinkFunction} />);
+
+  const backLink = screen.getByRole("button", { name: "Back" });
+  fireEvent.mouseDown(backLink);
+
+  expect(backLink).toBeFocused();
+});
+
+/* styling test for coverage */
```

Comment:
**suggestion** (non-blocking): maybe move the "just for coverage" tests to the very end, so it's clear when you get to them that there's nothing else important, and it's easier to remove them later if we can figure out how to get combined coverage.